sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_sender.py

```diff
@@ -446,28 +446,31 @@ def send_messages(
         except TypeError:   # continue if ServiceBusMessage
             pass
 
-        obj_message: Union[ServiceBusMessage, ServiceBusMessageBatch]
-
+        obj_message: Union[ServiceBusMessage, List[ServiceBusMessage]]
+        batch_message: ServiceBusMessageBatch
         if isinstance(message, ServiceBusMessageBatch):
             # If AmqpTransports are not the same, create batch with correct BatchMessage.
             if self._amqp_transport.KIND != message._amqp_transport.KIND: # pylint: disable=protected-access
                 # pylint: disable=protected-access
                 batch = self.create_message_batch()
-                batch._from_list(message._messages)  # type: ignore
-                obj_message = batch
+                batch._from_list(message._messages)
+                batch_message = batch
             else:
-                obj_message = message
+                batch_message = message
         else:
-            obj_message = transform_outbound_messages(  # type: ignore
+            obj_message = transform_outbound_messages(
                 message, ServiceBusMessage, self._amqp_transport.to_outgoing_amqp_message
             )
             try:
                 batch = self.create_message_batch()
-                batch._from_list(obj_message)  # type: ignore # pylint: disable=protected-access
-                obj_message = batch
+                # cast to iterable to avoid mypy error
+                obj_message = cast(List[ServiceBusMessage], obj_message)
+                batch._from_list(obj_message) # pylint: disable=protected-access
+                batch_message = batch
```

Comment:
EAFP and typing don't go together super well IMO. IIUC we're casting to list so L468 won't yell, but we don't actually know that it's a list right (b/c it might throw a TypeError)? We could refactor here, but I don't know this code well enough, so I leave up to you